projects/packages/autoloader/src/class-latest-autoloader-guard.php

```diff
@@ -45,38 +45,31 @@ public function __construct( $plugins_handler, $autoloader_handler, $autoloader_
 	 * has the side-effect of actually loading the latest autoloader in the event that this
 	 * is not it.
 	 *
-	 * @param string   $current_plugin The current plugin we're checking.
-	 * @param string[] $plugins        The active plugins to check for autoloaders in.
+	 * @param string   $current_plugin             The current plugin we're checking.
+	 * @param string[] $plugins                    The active plugins to check for autoloaders in.
+	 * @param bool     $was_included_by_autoloader Indicates whether or not this autoloader was included by another.
 	 *
 	 * @return bool True if we should stop initialization, otherwise false.
 	 */
-	public function should_stop_init( $current_plugin, $plugins ) {
-		global $jetpack_autoloader_including_latest;
+	public function should_stop_init( $current_plugin, $plugins, $was_included_by_autoloader ) {
 		global $jetpack_autoloader_latest_version;
 
-		// When we're being included from an older autoloader we need to
-		// reset the latest version so that the new autoloader can look
-		// for the latest autoloader again.
-		if ( $jetpack_autoloader_including_latest ) {
-			$jetpack_autoloader_latest_version = null;
```

Comment:
I removed this aspect of the check based on the observation that we really don't want to consider older autoloaders as eligible. We don't clear `$latest_version` in `Autoloader_Locator::find_latest_autoloader()` and so it will use whatever the existing latest version is as the minimum check attempting to find one newer.